### ISSUE TITLE
feature(kubernetes): Support for installing applications

### ISSUE THREAD

**Issue by dipankardas011** [kubesimplify/ksctl#153]
# Is your feature request related to a problem?📖
Currently ksctl only creates and deletes clusters. So a nice addition would be to have feature to add applications which will make the cluster ready for prod use

# Describe the solution you'd like 👨‍💻👩‍💻
installing pre-installed applications when the cluster is created and also when the cluster has already been created

# Describe alternatives you've considered💠
manual creation is not what ksctl's goal which is simplify and make it a good platform for anyone to manage clusters

# Provider in question
- [x] Civo
- [x] Local
- [x] Azure
- [x] K3s
- [x] Kubeadm

## Additional context
Check [Contribution's guidelines](https://kubesimplify.github.io/ksctl/docs/contributions)

**Comment by SD-13** [comment]
I would love to learn more about it and work on this issue. @dipankardas011 Can you please assign me. Thanks!

**Comment by dipankardas011** [comment]
also if pulumi can help we can use it 
https://www.pulumi.com/registry/packages/kubernetes/

**Comment by dipankardas011** [comment]
@prateek041 do have a close look at this issue ;>

**Comment by dipankardas011** [comment]
> I would love to learn more about it and work on this issue. @dipankardas011 Can you please assign me. Thanks!

Found out some way to apply cards via go client

**Comment by dipankardas011** [comment]
@SD-13 Here is quick way for you to get started
https://gitlab.com/dipankardas011/go-playground/-/tree/main/k8s-go
I hope it helps

also it does not represent how the ksctl should work with this but it helps you better understand and do some stuffs

**Comment by SD-13** [comment]
Certainly! Thanks @dipankardas011 I will start looking into them tonight and will reach out here if I have any doubts

**Comment by dipankardas011** [comment]
@SD-13 are you working on this issue?

**Comment by dipankardas011** [comment]
as this is a high priority feature proposal!

**Comment by SD-13** [comment]
Sorry for the delay, somehow it fell off my radar. I am starting working on it as a high priority. Thanks for checking btw!

**Comment by dipankardas011** [comment]
Sure @SD-13, np you can start
Also wanted to provide you with some context that the client for respective kubernetes API are added only thing is adding missing kubernetes pre-installed resources/kind and also how to handle all the `kubectl apply -f <URL>`

**Comment by dipankardas011** [comment]
also you can join the google group and discord server for latest development messages

**Comment by dipankardas011** [comment]
@SD-13 please migrate to this issue #179 closing this one!
